### PR TITLE
WIP LinShareExtension

### PR DESCRIPTION
**Pull request by aduprat**
It should work; but it doesn't...

The problem is that the networks can't be started.
```
STDERR: Creating network "3wd9zqdhvajz_linshare-tomcat-tier" with driver "bridge"
...
```

I've tried with two version of docker-compose: 1.22.0 & 1.8.0
I also played with the parameters: withPull & withLocalCompose

The same docker-compose elements (yml + configuration files) is working outside of TestContainers.

### CODE REVIEW

File: third-party/linshare/src/test/resources/conf/home/vendor/font-awesome/scss/_stacked.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Stacked Icons
+// -------------------------
+
+.#{$fa-css-prefix}-stack {
```
Comment (chibenwa) [inline third-party/linshare/src/test/resources/conf/home/vendor/font-awesome/scss/_stacked.scss]: We really have to include all the Linshare website resources in an Apache project?

File: third-party/pom.xml [hunk 2]
```diff
@@ -35,6 +35,7 @@
 
     <modules>
         <module>spamassassin</module>
+        <module>linshare</module>
```
Comment (chibenwa) [inline third-party/pom.xml]: alphabetic ordering?

**Comment by chibenwa** [comment]
```
[5e6bd3a10e194f65f0d2219422f1d8d898363c1a] [INFO] apache-james-linshare 3.2.0-SNAPSHOT ............... FAILURE [ 17.513 s]

[5e6bd3a10e194f65f0d2219422f1d8d898363c1a] 2018-09-03 17:03:50,858 ERROR [main] - [🐳 [docker/compose:1.8.0]]- Could not start container
[5e6bd3a10e194f65f0d2219422f1d8d898363c1a] java.lang.IllegalStateException: Container did not start correctly.
[5e6bd3a10e194f65f0d2219422f1d8d898363c1a] 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:272)
[5e6bd3a10e194f65f0d2219422f1d8d898363c1a] 	at 
```

File: third-party/linshare/src/test/java/org/apache/james/linshare/LinshareExtension.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class LinshareExtension implements BeforeAllCallback, AfterAllCallback, ParameterResolver {
+
+    public static final String BASE_PATH = "/linshare/webservice/rest/user/v2";
+    public static final String LINSHARE_UI_ADMIN_CONTAINER_NAME = "linshare_ui-admin";
+    public static final int LINSHARE_UI_ADMIN_PORT = 8080;
```
Comment (trantienduchn) [inline third-party/linshare/src/test/java/org/apache/james/linshare/LinshareExtension.java]: I checked, it exposed port 80 actually, if we want to use rest-assured then we need to exposes this port to host machine by declaring `ports` in `ui-admin` service from `docker-compose.yml`.

**Comment by mbaechler** [comment]
In my opinion, it doesn't make sense to use this docker compose file because it clearly make choices that doesn't fit with testing.
We should either create a new one or just run containers with testcontainers.

**Comment by trantienduchn** [comment]
getting a completed `linshare` image is cheaper than a dozen of `linshare` related images, and we can get rid of many resource files, those might be maintained in the future.

**Pull request closed without merging**